app/extras/username_generator.rb

```diff
@@ -0,0 +1,27 @@
+UsernameGenerator = Struct.new(:user) do
```

Comment:
I'm a Struct fan in instances where the initialize method is just boilerplate.

So having to write

```
attr_accessor :some_thing, :some_other_thing, :some_third_thing
def initialize(some_thing, some_other_thing, some_third_thing)
  @some_thing = some_thing
  @some_other_thing = some_other_thing
  @some_third_thing = some_third_thing
end
```
 is kinda verbose and unnecessary and I try to avoid it. (here it's only one thing, but I still like Structs for it)

Maybe it'd be useful to write a thing which made this work:

```
class UsernameGenerator
  initialize :some_thing, :some_other_thing, :some_third_thing
end
```